noir-projects/noir-contracts/contracts/app/private_token_contract/src/main.nr

```diff
@@ -1,46 +1,101 @@
+use aztec::macros::aztec;
 
+/// This is a toy example token used to demonstrate SinglePrivateMutable.
 #[aztec]
 pub contract PrivateToken {
-    use dep::aztec::macros::{functions::{external, initializer}, storage::storage};
-    use dep::aztec::messages::message_delivery::MessageDelivery;
-    use dep::aztec::protocol_types::address::AztecAddress;
-    use dep::aztec::state_vars::Owned;
-    use dep::balance_set::BalanceSet;
+    use aztec::{
+        macros::{functions::{external, initializer}, storage::storage},
+        messages::message_delivery::MessageDelivery,
+        protocol_types::address::AztecAddress,
+        state_vars::{Owned, SinglePrivateMutable},
+    };
+
+    use address_note::address_note::AddressNote;
+    use balance_set::BalanceSet;
+    use uint_note::uint_note::UintNote;
 
     #[storage]
     struct Storage<Context> {
         balances: Owned<BalanceSet<Context>, Context>,
+        // This contract has only a single admin whose identity we don't want to leak to the public so we store it in
+        // a single private mutable.
+        admin: SinglePrivateMutable<AddressNote, Context>,
+        // We don't want this contract to leak information about its total supply so we store it in a single private
+        // mutable.
+        total_supply: SinglePrivateMutable<UintNote, Context>,
     }
 
     /**
      * initialize the contract's initial state variables.
      */
     #[external("private")]
     #[initializer]
-    fn constructor(initial_supply: u64, owner: AztecAddress) {
-        self.storage.balances.at(owner).add(initial_supply as u128).deliver(
+    fn constructor(initial_admin_balance: u128, admin: AztecAddress) {
+        // Admin itself is the owner of the underlying note because we want admin to be able to transfer the admin
+        // privileges to someone else which requires being able to nullify the note. We use constrained onchain message
+        // delivery because we don't know if the party deploying this contract is incentivized to deliver the note.
+        self.storage.admin.initialize(AddressNote { address: admin }, admin).deliver(
+            MessageDelivery.CONSTRAINED_ONCHAIN,
+        );
+
+        // Mint the initial admin balance to the admin.
+        self.storage.balances.at(admin).add(initial_admin_balance).deliver(
             MessageDelivery.CONSTRAINED_ONCHAIN,
         );
+        self
+            .storage
+            .total_supply
+            .initialize(UintNote { value: initial_admin_balance }, admin)
+            .deliver(MessageDelivery.CONSTRAINED_ONCHAIN);
     }
 
     // Mints `amount` of tokens to `owner`.
     #[external("private")]
-    fn mint(amount: u64, owner: AztecAddress) {
-        self.storage.balances.at(owner).add(amount as u128).deliver(
-            MessageDelivery.CONSTRAINED_ONCHAIN,
-        );
+    fn mint(amount: u128, recipient: AztecAddress) {
+        let replacement_note_message = self.storage.admin.get_note();
```

Comment:
`get_note` returning a note message feels weird (this is shared with `PrivateMutable` but surprisingly we don't ever seem to use the `get_note` method of `PrivateMutable` in contracts now).

Reply:
[Discussed on slack](https://aztecprotocol.slack.com/archives/C07R3GTKJ49/p1765985821862809) and will potentially be addressed in a separate PR.